docs/cloud-docker-images.md

```diff
@@ -0,0 +1,100 @@
+5. Replace the python binary target for your launch script with
+`sematic_pipeline`, using the same `deps` as you use for the binary target
+6. Fill out the `registry` and `repository` fields of the `sematic_pipeline`
+with information about where to push your image.
+
+With that, you're done! Assuming the target for your launch script was
+`//my_repo/my_package:my_target`, you now have the following targets available:
+
+- `//my_repo/my_package:my_target`: still runs your target, but builds a Docker
+image with your code and its dependencies first
+- `//my_repo/my_package:my_target_local`: runs your target WITHOUT building and
+pushing the image. This can help for local development when you don't want the
+overhead of waiting for the build & push.
+- `//my_repo/my_package:my_target_image`: builds the image, but doesn't push it
+or run your script
+- `//my_repo/my_package:my_target_push`: builds and pushes the image, but
+doesn't run your script
+
+#### Custom base images
+The `sematic_pipeline` macro also allows you to specify a custom base image to
+cover any dependencies you have that aren't specified in bazel. You can do this
+by setting the `base` field of the `sematic_pipeline` macro, If, for
+example, your bazel setup assumes that it is running on a machine with a
+particular version of Cuda, you can bake that into the base image but let
+Sematic's bazel hookup handle all the python and other native code you want in
+the image. Sematic also provides a couple base images out-of-the-box, which can
+be referenced as `"@sematic-worker-base//image"` and
+`"@sematic-worker-cuda//image"` in bazel. `sematic-worker-cuda` includes
+python 3.9 and an installation of cuda, and is actually the base image used by
+default when you use the `sematic_pipeline` macro. `sematic-worker-base` is the
+same, but doesn't include cuda (this is helpful if you want a lighter image and
+don't use GPUs in your pipeline).
+
+If you like, you can also specify another base image not provided by Sematic.
+There are a few requirements on this image though, so be sure to read through
+the ["Totally Custom Image"](#totally-custom-image) section below to understand
+what those are. Note that if you are using bazel, requirements 1 & 2 from that
+section will already be taken care of.
+
+### requirements.txt
+Coming soon!
+
+### Totally Custom Image
+If you want full control over how your Docker image is produced, Sematic
+provides a hook to make that possible. Just set the `SEMATIC_CONTAINER_IMAGE`
+environment variable to the URI for the Docker image you want your pipeline to
+use. There are some requirements on this image though:
+
+1. It must contain your source code and its dependencies
+2. It must be pushed to a container registry that can be accessed by the cluster
+where your code is going to run in the cloud
+3. `/usr/bin/python3` must be a valid python interpreter in the image
+4. The home directory inside the image must be writable
```

Comment:
What about the entry point of the image? Do we override it in the K8s job spec anyway?

Reply:
Good point. This is another thing that's taken care of if you use our bazel macro.